src/client/token_provisioning/mod.rs

```diff
@@ -144,6 +169,78 @@ impl<'a> FobnailClient<'a> {
             return;
         }
 
-        todo!()
+        match state {
+            State::RequestPoCertChain { .. } => {
+                if result.header.code == MessageClass::Response(ResponseType::Content) {
+                    *state = State::VerifyPoCertChain {
+                        chain: result.payload,
+                    }
+                } else {
+                    error!("Server gave invalid response to certificate chain request");
+                    state.error();
+                }
+            }
+            _ => {
+                // We already matched all possible states in handle_response()
+                // and we should never reach here
+                unreachable!()
+            }
+        }
+    }
+
+    /// Verify certchain. On success certificates are loaded into certstore as
+    /// volatile certificates so that they can be used later.
+    fn verify_certchain<T>(trussed: &mut T, certmgr: &mut CertMgr, chain: &[u8]) -> Result<(), ()>
+    where
+        T: trussed::client::FilesystemClient + trussed::client::Sha256,
+    {
+        /// Minimal number of certificates required in a chain (including root).
+        const MIN_CERTS: usize = 2;
+        /// Max number of certificates allowed in a chain (including root).
+        const MAX_CERTS: usize = 3;
```

Comment:
If we want to use the same function later for checking whether newly created Fobnail certificate is valid, this should be 4.

Reply:
@krystian-hebel We won't use the same function, PO chain is still loaded and we only need to call `certmgr.verify()`